Ctrl+D to exit moderated session results in server panic

Expected behavior:
You should be able to use Ctrl + D to exit a moderated session from the web UI.

Current behavior:
Doing so currently results in `master` results in a panic.  The end logs of the server are as follows:
```
2023-02-06T12:10:39-07:00 DEBU [CLIENT]    Error copying data to remote peer: read tcp 192.168.1.201:443->192.168.1.166:51112: use of closed network connection client/session.go:693
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Sent close event to web client. session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:722
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Sent close event to web client. session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:722
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Terminating websocket ping loop. session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:367
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Closing websocket stream session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:424
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Closing websocket stream session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:424
2023-02-06T12:10:39-07:00 DEBU [WEBSOCKET] Terminating websocket ping loop. session_id:a4050684-f913-47d3-a5be-7a823d278a2d web/terminal.go:367
panic: repeated read on failed websocket connection

goroutine 2014 [running]:
github.com/gorilla/websocket.(*Conn).NextReader(0xc002520c60)
	github.com/gorilla/websocket@v1.5.0/conn.go:1030 +0x2a8
github.com/gorilla/websocket.(*Conn).ReadMessage(0x76315e0?)
	github.com/gorilla/websocket@v1.5.0/conn.go:1093 +0x19
github.com/gravitational/teleport/lib/web.(*TerminalStream).Read(0xc002519040, {0xc002020aa4, 0x1, 0xc001f4a7b0?})
	github.com/gravitational/teleport/lib/web/terminal.go:1013 +0x6f
github.com/gravitational/teleport/lib/client.handleNonPeerControls({0xc0025ae190, 0x9}, 0xc0023b4ba0, 0xc002007fa8)
	github.com/gravitational/teleport/lib/client/session.go:646 +0x6b
github.com/gravitational/teleport/lib/client.(*NodeSession).pipeInOut.func2()
	github.com/gravitational/teleport/lib/client/session.go:718 +0xa5
created by github.com/gravitational/teleport/lib/client.(*NodeSession).pipeInOut
	github.com/gravitational/teleport/lib/client/session.go:715 +0x245
```

--- a/lib/web/terminal.go
+++ b/lib/web/terminal.go
@@ -1012,7 +1012,10 @@ func (t *TerminalStream) Read(out []byte) (n int, err error) {
 
 	ty, bytes, err := t.ws.ReadMessage()
 	if err != nil {
-		if err == io.EOF || websocket.IsCloseError(err, websocket.CloseAbnormalClosure, websocket.CloseGoingAway, websocket.CloseNormalClosure) {
+		// if the connection has closed, we must return io.EOF in order to abort
+		// the websocket copy loop
+		if err == io.EOF || errors.Is(err, net.ErrClosed) ||
+			websocket.IsCloseError(err, websocket.CloseAbnormalClosure, websocket.CloseGoingAway, websocket.CloseNormalClosure) {
 			return 0, io.EOF
 		}
 

